crates/isograph_compiler/src/refetch_fields.rs

```diff
@@ -43,9 +43,11 @@ fn add_refetch_field_to_object(
         Entry::Vacant(vacant_entry) => {
             let next_client_field_id = client_fields.len().into();
 
-            vacant_entry.insert(FieldType::ClientField(next_client_field_id));
+            vacant_entry.insert(FieldType::ClientField(ClientType::ClientField(
+                next_client_field_id,
+            )));
 
-            client_fields.push(ClientField {
+            client_fields.push(ClientType::ClientField(ClientField {
```

Comment:
Not in this PR, but long term, perhaps we can get rid of variant by making refetch (etc) fields `FieldType::Server`? Or something. Because `FieldType::Client` implies that there's a selection set, it's user generated, etc